Build fails on Arch Linux unless PYTHON env. variable is set

`cargo build` bails out with 
```
--- stderr
  File "/home/marcin/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.50.0/mozjs/js/src/../../configure.py", line 83
    os.chmod('config.status', 0755)
                                 ^
SyntaxError: invalid token
cat: config.log: No such file or directory
make: *** [makefile.cargo:140: maybe-configure] Error 1
thread 'main' panicked at 'assertion failed: result.success()', /home/marcin/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.50.0/build.rs:53
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```
No such error occurs if the invocation is `PYTHON=python2 cargo build`.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,7 +2,7 @@
 name = "mozjs_sys"
 description = "System crate for the Mozilla SpiderMonkey JavaScript engine."
 repository = "https://github.com/servo/mozjs/"
-version = "0.50.0"
+version = "0.50.1"
 authors = ["Mozilla"]
 links = "mozjs"
 build = "build.rs"

--- a/makefile.cargo
+++ b/makefile.cargo
@@ -50,7 +50,14 @@ ifeq (,$(VCINSTALLDIR))
 		CXX ?= g++
 	endif
 AR ?= ar
+
+# check if python2 is a valid Python executable, otherwise fall back to python
+ifeq (, $(findstring Python 2.,$(shell python2 --version 2> /dev/null)))
+PYTHON ?= python2
+else
 PYTHON ?= python
+endif 
+
 endif
 
 endif

